cloud/ec2_fleet.go

```diff
@@ -0,0 +1,538 @@
+		statuses = append(statuses, statusMap[h.Id])
+	}
+	return statuses, nil
+}
+
+func (m *ec2FleetManager) GetInstanceStatus(ctx context.Context, h *host.Host) (CloudStatus, error) {
+	r, err := getRegion(h)
+	if err != nil {
+		return StatusUnknown, errors.Wrap(err, "problem getting region from host")
+	}
+	if err = m.client.Create(m.credentials, r); err != nil {
+		return StatusUnknown, errors.Wrap(err, "error creating client")
+	}
+	defer m.client.Close()
+
+	instance, err := m.getInstance(ctx, h)
+	if err != nil {
+		return StatusUnknown, errors.Wrapf(err, "can't get instance status")
+	}
+
+	return ec2StatusToEvergreenStatus(*instance.State.Name), nil
+}
+
+func (m *ec2FleetManager) TerminateInstance(ctx context.Context, h *host.Host, user string) error {
+	if h.Status == evergreen.HostTerminated {
+		return errors.Errorf("Can not terminate %s - already marked as terminated!", h.Id)
+	}
+	r, err := getRegion(h)
+	if err != nil {
+		return errors.Wrap(err, "problem getting region from host")
+	}
+	if err = m.client.Create(m.credentials, r); err != nil {
+		return errors.Wrap(err, "error creating client")
+	}
+	defer m.client.Close()
+
+	resp, err := m.client.TerminateInstances(ctx, &ec2.TerminateInstancesInput{
+		InstanceIds: []*string{aws.String(h.Id)},
+	})
+	if err != nil {
+		grip.Error(message.WrapError(err, message.Fields{
+			"message":       "error terminating instance",
+			"user":          user,
+			"host":          h.Id,
+			"host_provider": h.Distro.Provider,
+			"distro":        h.Distro.Id,
+		}))
+		return err
+	}
+
+	for _, stateChange := range resp.TerminatingInstances {
+		if stateChange == nil || stateChange.InstanceId == nil {
+			grip.Error(message.Fields{
+				"message":       "state change missing instance ID",
+				"user":          user,
+				"host_provider": h.Distro.Provider,
+				"host":          h.Id,
+				"distro":        h.Distro.Id,
+			})
+			return errors.New("invalid terminate instances response")
+		}
+		grip.Info(message.Fields{
+			"message":       "terminated instance",
+			"user":          user,
+			"host_provider": h.Distro.Provider,
+			"instance_id":   *stateChange.InstanceId,
+			"host":          h.Id,
+			"distro":        h.Distro.Id,
+		})
+	}
+
+	return errors.Wrap(h.Terminate(user), "failed to terminate instance in db")
+}
+
+func (m *ec2FleetManager) IsUp(ctx context.Context, h *host.Host) (bool, error) {
+	status, err := m.GetInstanceStatus(ctx, h)
+	if err != nil {
+		return false, errors.Wrap(err, "error checking if instance is up")
+	}
+	if status == StatusRunning {
+		return true, nil
+	}
+	return false, nil
+}
+
+// OnUp is a noop for ec2Fleet
+func (m *ec2FleetManager) OnUp(context.Context, *host.Host) error {
+	return nil
+}
+
+func (m *ec2FleetManager) GetDNSName(ctx context.Context, h *host.Host) (string, error) {
+	r, err := getRegion(h)
+	if err != nil {
+		return "", errors.Wrap(err, "problem getting region from host")
+	}
+	if err = m.client.Create(m.credentials, r); err != nil {
```

Comment:
Are calls to `m.client.Create` redundant? `GetManager` calls `provider.Configure`.

Reply:
I wanted to put client.Create in the individual calls instead of in `Configure` so I could manage returning the client to the pool (this allowed to me call `client.Close()`)